practice_app/api/views.py

```diff
@@ -543,3 +557,35 @@ def post_papers(request):
         paper.like_count = 0
         paper.save() # save to the DB
     return JsonResponse({'status': 'Requested papers are saved successfully.'}, status=200) # success response
+
+
+@csrf_exempt
+def add_paper_to_list(request):
+    if request.user.is_anonymous:
+        if 'username' not in request.headers or 'password' not in request.headers:
+            return JsonResponse({'status': 'username and password fields can not be empty '}, status=407)
+        username = request.headers['username']
+        password = request.headers['password']
+        user = authenticate(request, username=username, password=password)
+        if user == None:
+            return JsonResponse({'status' : 'user credentials are incorrect.'},status=401)
+    
+    data = request.POST
+    list_id = data.get("list_id")
+    pid = data.get("paper_id")
```

Comment:
I think you should handle the case where these are not given.

Reply:
OK, I added an if condition to check it